src/particles/elements/mixin/alignment.H

```diff
@@ -0,0 +1,146 @@
+     */
+    struct Alignment
+    {
+        static constexpr amrex::ParticleReal degree2rad = ablastr::constant::math::pi / 180.0;
+
+        /** A finite-length element
+         *
+         * @param dx horizontal translation error in m
+         * @param dy vertical translation error in m
+         * @param rotation_degree transversal rotation error around the z axis (in the x-y plane) [degrees]
+         */
+        Alignment (
+            amrex::ParticleReal dx,
+            amrex::ParticleReal dy,
+            amrex::ParticleReal rotation_degree
+        )
+        : m_dx(dx), m_dy(dy), m_rotation(rotation_degree * degree2rad)
+        {
+        }
+
+        Alignment () = default;
+        Alignment (Alignment const &) = default;
+        Alignment& operator= (Alignment const &) = default;
+        Alignment (Alignment&&) = default;
+        Alignment& operator= (Alignment&& rhs) = default;
+
+        ~Alignment () = default;
+
+        /** Shift the particle into the alignment error frame
+         *
+         * @param[inout] x horizontal position relative to reference particle
+         * @param[inout] y vertical position relative to reference particle
+         * @param[inout] px horizontal momentum relative to reference particle
+         * @param[inout] py vertical momentum relative to reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void shift_in (
+            amrex::ParticleReal & AMREX_RESTRICT x,
+            amrex::ParticleReal & AMREX_RESTRICT y,
+            amrex::ParticleReal & AMREX_RESTRICT px,
+            amrex::ParticleReal & AMREX_RESTRICT py
+        ) const
+        {
+            auto const [sin_rotation, cos_rotation] = amrex::Math::sincos(m_rotation);
+
+            // position
+            x -= m_dx;
+            y -= m_dy;
+            x = x * cos_rotation + y * sin_rotation;
+            y = -x * sin_rotation + y * cos_rotation;
+
+            // momentum
+            px = px * cos_rotation + py * sin_rotation;
+            py = -px * sin_rotation + py * cos_rotation;
+        }
+
+        /** Shift the particle out of the alignment error frame
+         *
+         * @param[inout] x horizontal position relative to reference particle
+         * @param[inout] y vertical position relative to reference particle
+         * @param[inout] px horizontal momentum relative to reference particle
+         * @param[inout] py vertical momentum relative to reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void shift_out (
+            amrex::ParticleReal & AMREX_RESTRICT x,
+            amrex::ParticleReal & AMREX_RESTRICT y,
+            amrex::ParticleReal & AMREX_RESTRICT px,
+            amrex::ParticleReal & AMREX_RESTRICT py
+        ) const
+        {
+            auto const [sin_rotation, cos_rotation] = amrex::Math::sincos(-m_rotation);
```

Comment:
I think changing the sign here in the naming of sincos is confusing.  I recommend changing the signs inside the transformation, or calling this something different.

Reply:
We can do that, sure. Just to understand: why is it confusing? It is rotating forward/backward by x degrees :sweat_smile: